src/coreclr/tools/aot/ILCompiler.ReadyToRun/Compiler/ReadyToRunMetadataFieldLayoutAlgorithm.cs

```diff
@@ -606,7 +606,13 @@ public FieldAndOffset[] GetOrAddDynamicLayout(DefType instantiatedType, FieldAnd
 
         protected override ComputedInstanceFieldLayout ComputeInstanceFieldLayout(MetadataType type, int numInstanceFields)
         {
-            if (type.IsExplicitLayout)
+            ClassLayoutMetadata layoutMetadata = type.GetClassLayout();
+            MetadataLayoutKind layoutKind = layoutMetadata.Kind;
+            if (layoutKind == MetadataLayoutKind.CStruct)
```

Comment:
Should this be switch? You can use `goto default;` to handle the SequentialLayout quirk.

```
    switch (layoutMetadata.Kind)
    {
        case MetadataLayoutKind.Sequential:
            if (type.ContainsGCPointers)
                goto default;
....
            return ComputeSequentialFieldLayout(type, numInstanceFields, layoutMetadata);
        case MetadataLayoutKind.Explicit
...
            return ComputeExplicitFieldLayout(type, numInstanceFields, layoutMetadata);
        case ...

        default:
            return ComputeAutoFieldLayout(type, numInstanceFields, layoutMetadata);
    }
```